ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op.cpp

```diff
@@ -72,9 +74,11 @@ Tensor optimized_conv_new(const Tensor& a, const Tensor &b, std::optional<const
     bool enable_subblock_padding,
     bool use_non_tile_height
 ) {
+    tt::tt_metal::Device* device = a.device();
+    auto stats = device->get_memory_allocation_statistics(tt::tt_metal::BufferType::L1);
```

Comment:
you are picking the stats here, and op is launched later on
Is it possible that something else is going to change the state of l1 memory in between these two things?

Reply:
Move pre op stats calc to within launch op

Reply:
Done.